tests/dualtor/test_server_failure.py

```diff
@@ -24,23 +24,27 @@ def test_server_down(duthosts, tbinfo, rand_selected_interface, simulator_flap_c
         
     upper_tor = duthosts[tbinfo['duts'][0]]
     lower_tor = duthosts[tbinfo['duts'][1]]
-    PAUSE_TIME = 5
+    
+    def upper_tor_mux_state_verification(state, health):
+        mux_state_upper_tor = show_muxcable_status(upper_tor)
+        return mux_state_upper_tor[itfs]['status'] == state and mux_state_upper_tor[itfs]['health'] == health
+    
+    def lower_tor_mux_state_verfication(state, health):
+        mux_state_lower_tor = show_muxcable_status(lower_tor)
+        return mux_state_lower_tor[itfs]['status'] == state and mux_state_lower_tor[itfs]['health'] == health
 
     itfs, _ = rand_selected_interface
     # Set upper_tor as active
     toggle_simulator_port_to_upper_tor(itfs)
-    time.sleep(PAUSE_TIME)
+    wait_until(10, 1, 0, upper_tor_mux_state_verification, 'active', 'healthy')
```

Comment:
can test continue if this step failed?

You can increase the timeout to 30 seconds, the point of using wait until is to let normal cases go fast and wait long enough to have a sure failure when failure happens.

Reply:
See your point now. Updated accordingly.

Reply:
There is still an outstanding question, if the toggling failed here, should we fail the test or proceed?

Reply:
Per offline discussion, we should fail the test here so the real cause of test failure wouldn't be hidden.